tests/operator/combine.ts

```diff
@@ -54,4 +56,20 @@ describe('Stream.prototype.combine', () => {
       },
     });
   });
+
+  it('should return a Stream when combining a MemoryStream with a Stream', (done) => {
+    const input1 = xs.periodic(50).take(4).remember();
+    const input2 = xs.periodic(80).take(3);
+    const stream: Stream<number> = input1.combine((x, y) => x + y, input2);
+    assert.strictEqual(stream instanceof Stream, true);
+    done();
+  });
+
+  it('should return a Stream when combining a MemoryStream with a MemoryStream', (done) => {
+    const input1 = xs.periodic(50).take(4).remember();
+    const input2 = xs.periodic(80).take(3).remember();
+    const stream: Stream<number> = input1.combine((x, y) => x + y, input2);
+    assert.strictEqual(stream instanceof Stream, true);
+    done();
+  });
```

Comment:
How does this behavior work? I don't see any adjustments to `combine()`, and I don't know how it would know how to do this